app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,13 @@
+module Foreman
+  class Deprecation
+    #deadline_version - is the version the deprecation is going to be deleted, the format must be a major release e.g "1.8"
+    def self.deprecation_warning(foreman_version_deadline, info)
+      raise Foreman::Exception.new(N_("Invalid version format, please enter in x.y (only major version).")) unless foreman_version_deadline.to_s.match(/\A\d[.]\d+\z/)
+      ActiveSupport::Deprecation.warn("You are using a deprecated behavior, it will be removed in version #{foreman_version_deadline}, #{info}", caller)
```

Comment:
That was actually a typo in my comment, I actually ran `Host.find_by_ip`, but in either case I don't believe Host::Managed overrides the method at all - it's just a normal class extension, there's only one find_by_ip method.  The message also says it's called from base.rb, not managed.rb.